src/utils/SpokeUtils.ts

```diff
@@ -20,36 +21,14 @@ export async function getBlockRangeForDepositId(
   initLow: number,
   initHigh: number,
   maxSearches: number,
-  spokePool: Contract,
+  spokePool: SpokePoolClient,
   deploymentBlock = 0
 ): Promise<{
   low: number;
   high: number;
 }> {
-  // Define a mapping of block numbers to number of deposits at that block. This saves repeated lookups.
-  const queriedIds: Record<number, number> = {};
-
-  // Define a llambda function to get the deposit ID at a block number. This function will first check the
-  // queriedIds cache to see if the deposit ID at the block number has already been queried. If not, it will
-  // make an eth_call request to get the deposit ID at the block number. It will then cache the deposit ID
-  // in the queriedIds cache.
-  const _getDepositIdAtBlock = async (blockNumber: number): Promise<number> => {
-    if (queriedIds[blockNumber] === undefined) {
-      queriedIds[blockNumber] = await getDepositIdAtBlock(spokePool, blockNumber);
-    }
-    return queriedIds[blockNumber];
-  };
-
-  // Get the most recent block from the spoke pool contract, the deposit ID
-  // at the low block, and the deposit ID at the high block in parallel.
-  const [mostRecentBlockNumber, highestPossibleDepositIdInRange, lowestPossibleDepositIdInRange] = await Promise.all([
-    spokePool.provider.getBlockNumber(),
-    _getDepositIdAtBlock(initHigh),
-    _getDepositIdAtBlock(Math.max(deploymentBlock, initLow - 1)),
-  ]);
-
   // Set the initial high block to the most recent block number or the initial high block, whichever is smaller.
-  initHigh = Math.min(initHigh, mostRecentBlockNumber);
+  initHigh = Math.min(initHigh, spokePool.latestBlockNumber);
```

Comment:
`spokePool` is guaranteed to be `update`ed right?

Reply:
Good call - resolved